ci/infra/azure/worker-instance.tf

```diff
@@ -3,28 +3,32 @@ resource "azurerm_network_interface" "worker" {
   name                = "${var.stack_name}-worker-${count.index}-nic"
   location            = azurerm_resource_group.resource_group.location
   resource_group_name = azurerm_resource_group.resource_group.name
+  depends_on          = [azurerm_subnet.nodes]
 
   ip_configuration {
     name                          = "internal"
     subnet_id                     = azurerm_subnet.nodes.id
     private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.create_bastionhost ? null : element(azurerm_public_ip.worker.*.id, count.index)
```

Comment:
I think we have to drop that, is it here just for your debugging purposes? We don't want to add public IPs to the worker nodes when everything is ready to be shipped.

Reply:
yes. this will be removed once finish verifying